pkg/operator/controller/ingress/deployment_test.go

```diff
@@ -1109,6 +1115,115 @@ func TestDesiredRouterDeploymentClientTLS(t *testing.T) {
 	}
 }
 
+// TestDesiredRouterDeploymentDynamicConfigManager verifies that desiredRouterDeployment
+// returns the expected deployment when DCM featuregate is involved.
+func TestDesiredRouterDeploymentDynamicConfigManager(t *testing.T) {
+	testCases := []struct {
+		name                       string
+		unsupportedConfigOverrides string
+		dcmEnabled                 bool
+		expectedEnv                []envData
+	}{
+		{
+			name:       "featuregate-enabled-default-values",
+			dcmEnabled: true,
+			expectedEnv: []envData{
+				{"ROUTER_HAPROXY_CONFIG_MANAGER", true, "true"},
+				{"ROUTER_MAX_DYNAMIC_SERVERS", true, "1"},
+				{"ROUTER_BLUEPRINT_ROUTE_POOL_SIZE", true, "0"},
+			},
+		},
+		{
+			name:                       "featuregate-enabled-max-servers",
+			unsupportedConfigOverrides: `{"maxDynamicServers":"7"}`,
+			dcmEnabled:                 true,
+			expectedEnv: []envData{
+				{"ROUTER_HAPROXY_CONFIG_MANAGER", true, "true"},
+				{"ROUTER_MAX_DYNAMIC_SERVERS", true, "7"},
+				{"ROUTER_BLUEPRINT_ROUTE_POOL_SIZE", true, "0"},
+			},
+		},
+		{
+			name:                       "featuregate-enabled-max-servers-lower-limit",
+			unsupportedConfigOverrides: `{"maxDynamicServers":"0"}`,
+			dcmEnabled:                 true,
+			expectedEnv: []envData{
+				{"ROUTER_HAPROXY_CONFIG_MANAGER", true, "true"},
+				{"ROUTER_MAX_DYNAMIC_SERVERS", true, "1"},
+				{"ROUTER_BLUEPRINT_ROUTE_POOL_SIZE", true, "0"},
+			},
+		},
+		{
+			name:                       "featuregate-enabled-dcm-override",
+			unsupportedConfigOverrides: `{"dynamicConfigManager":"false","maxDynamicServers":"7"}`,
+			dcmEnabled:                 true,
+			expectedEnv: []envData{
+				{"ROUTER_HAPROXY_CONFIG_MANAGER", false, ""},
+				{"ROUTER_MAX_DYNAMIC_SERVERS", false, ""},
+				{"ROUTER_BLUEPRINT_ROUTE_POOL_SIZE", false, ""},
+			},
+		},
+		{
+			name:       "featuregate-disabled-default-values",
+			dcmEnabled: false,
+			expectedEnv: []envData{
+				{"ROUTER_HAPROXY_CONFIG_MANAGER", false, ""},
+				{"ROUTER_MAX_DYNAMIC_SERVERS", false, ""},
+				{"ROUTER_BLUEPRINT_ROUTE_POOL_SIZE", false, ""},
+			},
+		},
+		{
+			name:                       "featuregate-disabled-max-servers",
+			unsupportedConfigOverrides: `{"maxDynamicServers":"7"}`,
+			dcmEnabled:                 false,
+			expectedEnv: []envData{
+				{"ROUTER_HAPROXY_CONFIG_MANAGER", false, ""},
+				{"ROUTER_MAX_DYNAMIC_SERVERS", false, ""},
+				{"ROUTER_BLUEPRINT_ROUTE_POOL_SIZE", false, ""},
+			},
+		},
+		{
+			name:                       "featuregate-disabled-dcm-override",
+			unsupportedConfigOverrides: `{"dynamicConfigManager":"true","maxDynamicServers":"100"}`,
+			dcmEnabled:                 false,
+			expectedEnv: []envData{
+				{"ROUTER_HAPROXY_CONFIG_MANAGER", true, "true"},
+				{"ROUTER_MAX_DYNAMIC_SERVERS", true, "100"},
+				{"ROUTER_BLUEPRINT_ROUTE_POOL_SIZE", true, "0"},
+			},
+		},
```

Comment:
Can you add a test to see what happens when `unsupportedConfigOverrides.DynamicConfigManager` is an empty string?

Reply:
`featuregate-enabled-default-values` and `featuregate-enabled-max-servers` tests don't set `unsupportedConfigOverrides.dynamicConfigManager` which results into an `""` after the unmarshaling.

Reply:
Sorry, I meant when the max dynamic servers is set to "" or something like "100K" or over the HAProxy limit for numbers.